### ISSUE TITLE
Collapsing Bookmarks within tags and displaying ayah info

### ISSUE THREAD

**Issue by MalikSaabSays** [quran/quran_android#683]
Salam all.

I've accumulated a lot of ayah bookmarks under many tags. When viewed under the bookmarks tab there is an endless list and is a pain to scroll through.

1. Can we have a way to display tags as collapsible items? Much like the [+] that one clicks to expand and collapse options on the PC. A button to collapse all tags could be placed under the menu where it says 'Group by Tags'.

2. Can we have the option to display additional selectable information in the ayah bookmarks, like the first few words from a translation that the end-user can select in options to have something like up to a line of text as a preview when viewing in the bookmarks tab. 

What do you guys think?

**Comment by ahmedre** [comment]
wa3laikum alsalam,
so just to make sure i understand you correctly, for 1, you want a way to make tags expandable/collapsable - right?

2 is interesting - i wanted that with some preview of the Arabic text even. will think about it, it shouldn't be much extra work to make it read from the active translation instead if we did make it read from the Arabic.

**Comment by MalikSaabSays** [comment]
Yes, it is exactly as you've put it - expandable/collapsable tags.

**Comment by MalikSaabSays** [comment]
Just had another thought.
Like individual notifications on android can be expanded by using two-finger gesture (similar to zooming-in on a photo), individual a yah tags could be expanded to reveal further info (ayat itself/translation etc), for quick reference.

**Comment by MalikSaabSays** [comment]
Salam.

So it's been like 5 n a half years. Havn't seen any activity towards resolution. There have been more tickets since, by others: Tickets #1342 #1607.

Its become impossible to navigate bookmarks from the main screen. There are so many bookmarks that the scroll bar is like a dot on the very edge of the screen.

Suggested solution:
1. Expandable/collapsible categories of bookmarks. Ability to expand one category only while rest stay collapsed.
2. Have a vertical scrolling bar bubble. So we have something clear to tap and drag for scrolling through bookmarks.

If devs are busy, can I hire some people to do this? Do I just point them to this GitHub? How does it work? I am willing to contribute.

**Comment by MalikSaabSays** [comment]
@ahmedre can I hope for progress on this?
In the least, the scroll bar needs a bubble so I can scroll!

**Comment by ahmedre** [comment]
no progress yet, sorry :(
for a scroll bubble, you mean something like what we have in the sura list / juz' list not he index page?

**Comment by MalikSaabSays** [comment]
![images](https://user-images.githubusercontent.com/22034853/228720825-2d520879-f45c-4893-98e0-4602bf8b84ca.png)


The scroll bar on the side becomes very thin when there is a long list of bookmarks. A bubble on the side like in above image, that appears even you start scrolling, so our fingers know where to tap and drag for scrolling through the long list.

**Comment by MalikSaabSays** [comment]
![Screenshot_20230330_083646](https://user-images.githubusercontent.com/22034853/228722649-6e98996d-a7e1-48f8-9d27-ddda6d869513.jpg)


The problem can be seen here. The bar to tap and drag for scrolling is just so tiny that its impossible to scroll with it.
I'm suggesting an icon of the Quran app could appear as a bubble next to it once we start scrolling. One can then tap and drag on the icon.
It can also be a bubble having first letter of the Tag much like how the first letter of the series is displayed when scrolling in the phone's contacts list.

**Comment by MalikSaabSays** [comment]
> no progress yet, sorry :( for a scroll bubble, you mean something like what we have in the sura list / juz' list not he index page?

@ahmedre 
A bubble like this to hold and scroll with, will be mighty mighty helpful

![Screenshot_20230405_112106](https://user-images.githubusercontent.com/22034853/229997751-3a57cbc8-5e61-4431-b12d-53b9cc58610f.jpg)

**Comment by MalikSaabSays** [comment]
> this sounds reasonable
> 
> out of curiosity, why have so many bookmarks? is it a way of categorizing certain ayat or what is your thought behind it?

Well @ahmedre brother, as I read I categorize ayaat according to topics of interest for ready reference, much like a study Quran. The Quran structure is such that each topic is addressed multiple times from various angles and it isn't in one place but spread all over. So I bring ayaat that deal in one narrative, under one label.

Yes it does make that tab ultra-congested for me, and I navigate it every day. It'd be good to have a way to scroll through it, or have collapsible tags.

**Comment by MalikSaabSays** [comment]
If you need some help with the workload of coding, I can probably reach out to some relevant people. Entirely upto you.

**Comment by MalikSaabSays** [comment]
@ahmedre bump

**Comment by MalikSaabSays** [comment]
@ahmedre Any response would be great

**Comment by ahmedre** [comment]
will in sha' Allah do this when I re-do the bookmarks screen in the near future in sha' Allah.
